src/pages/index.js

```diff
@@ -0,0 +1,627 @@
+// Array of all modals aka pop ups
+const popUps = Array.from(page.querySelectorAll(".modal"));
+// Outside of any functions, declare variables to store the current selected
+// card and its ID. You will assign them values when a delete icon is clicked.
+let selectedCard;
+let selectedCardId;
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "263132f6-100d-422f-9deb-cb4a2234cfd1",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .renderUserInfo()
+  .then((result) => {
+    // process the result
+    // console.log(result);
+    setInitialProfile(result);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+function setInitialProfile(initialProfile) {
+  profileInfoSave[0].textContent = initialProfile.name;
+  profileInfoSave[1].textContent = initialProfile.about;
+  modalAvatarImage.src = initialProfile.avatar;
+}
+
+// Defining profile and newpost modals' variables
+const avatarInfo = page.querySelector(".avatar__information");
+const profileInfoSave = [
+  avatarInfo.querySelector(".avatar__name"),
+  avatarInfo.querySelector(".avatar__description"),
+];
+
+const profileInfoValue = [
+  modalEditProfile.querySelector("#name"),
+  modalEditProfile.querySelector("#description"),
+];
+
+const modalNewPostValue = [
+  modalNewPost.querySelector("#picture"),
+  modalNewPost.querySelector("#caption"),
+];
+
+const modalAvatarImage = profile.querySelector(".avatar__image");
+
+const modalEditProfileAvatarImageInput =
+  modalEditProfileAvatar.querySelector("#picture");
+
+// // Event listeners to toggle edit profile and newpost modals on and off
+editProfileButton.addEventListener("click", () => {
+  resetErrorDisplayedInEditProfileModal(modalEditProfile, settings);
+  disableSubmitButton(submitProfileButton);
+  openModal(modalEditProfile);
+
+  // TODO: Get the values of each form field from the value property
+  // of the corresponding input element.
+  for (let i = 0; i < profileInfoValue.length; i++) {
+    profileInfoValue[i].value = profileInfoSave[i].textContent;
+  }
+});
+
+editProfileOverlayedButton.addEventListener("click", () => {
+  resetErrorDisplayedInEditProfileModal(modalEditProfileAvatar, settings);
+  disableSubmitButton(formEditProfileAvatar);
+  openModal(modalEditProfileAvatar);
+
+  // TODO: Get the values of each form field from the value property
+  // of the corresponding input element.
+});
+// Handle submitted profile function
+function handleEditProfileAvatarFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  // notify the user that the upload process is underway
+  // by changing the button text to "Saving..."
+  editProfileAvatarSubmitButton.textContent = "Saving...";
+
+  // TODO: Then insert these new values into the src property of the
+  // profile image.
+  modalAvatarImage.src = modalEditProfileAvatarImageInput.value;
```

Comment:
`  modalAvatarImage.src = modalEditProfileAvatarImageInput.value;
`

You need to update the `DOM` only in blocks  `.then` after a successful response from the server, so that when there is an error, it shouldn’t change



**Please, correct it everywhere in the code**